### PR TITLE
Bug 1225494 - Clipboard pastes first frame from GIF images (copied from Firefox) into other applications

### PR DESCRIPTION
**Pull request by Johennes**

### CODE REVIEW

File: Client/Extensions/UIPasteboardExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import UIKit
+
```
Comment (fluffyemily) [inline Client/Extensions/UIPasteboardExtensions.swift]: You'll now need to add `#import MobileCoreServices` to the top here. So sorry for taking so long to get round to this.

File: ClientTests/UIPasteboardExtensionsTests.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import UIKit
+import XCTest
+
```
Comment (fluffyemily) [inline ClientTests/UIPasteboardExtensionsTests.swift]: needs `import MobileCoreServices`

File: Client/Extensions/UIPasteboardExtensions.swift [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+extension UIPasteboard {
+
+    func addImageWithData(data: NSData, forURL url: NSURL) {
+        let isGIF = UIImage.dataIsGIF(data)
+
+        guard let image = imageWithData(data, isGIF: isGIF) else {
```
Comment (fluffyemily) [inline Client/Extensions/UIPasteboardExtensions.swift]: turns out you do not need to turn the NSData into image. If you set the NSData directly with the gif key into the pasteboard then it pastes as animated.
```
let isGIF = UIImage.dataIsGIF(data)

        // Setting pasteboard.items allows us to set multiple representations for the same item.
        items = [[
            kUTTypeURL as String: url,
            imageTypeKey(isGIF): data
        ]]
```

**Comment by Johennes** [comment]
@fluffyemily: All done, thanks for reviewing. :)

**Pull request merged**